### ISSUE TITLE
cli "deployments" rationale (and _user_dir breaking docker)

### ISSUE THREAD

**Issue by sanzoghenzo** [mamba-org/quetz#216]
**TL;DR**: if I run `quetz run` in a docker container without a user directory, the container exits with a permission error while trying to create the directory `/nonexistent/.config/quetz` in which to place the `deployments,json` file.
I'm wondering if this config file is needed at all. 

This is my Dockerfile, that builds the latest git version of quetz and runs it with the option added with #212:

```dockerfile
FROM condaforge/miniforge3:latest as base

WORKDIR build
ARG CONDA_ENV=/opt/venv
COPY environment.yml .
RUN conda env create -p $CONDA_ENV -f environment.yml
COPY pyproject.toml setup.py LICENSE ./
COPY quetz ./quetz
RUN conda run -p $CONDA_ENV pip install . --no-deps

FROM gcr.io/distroless/base-debian10 as production
ARG CONDA_ENV=/opt/venv
ENV LANG=en_US.UTF-8
ENV PATH $CONDA_ENV/bin:$PATH
COPY --from=base $CONDA_ENV $CONDA_ENV
USER nobody
WORKDIR /deployment
ENTRYPOINT ["quetz"]
CMD ["run", ".", "--skip-if-exists"]
```

as you can see the working directory is set to `/deployment` and the default command tries to run the deployment in the current directory, if it finds a `config.toml` in it.

with a `docker-compose.yml` like this:
```yml
version: "3.8"

services:
  quetz:
    image: "quetz:latest"
    build:
      context: ..
      dockerfile: docker/Dockerfile
    container_name: quetz
    ports:
      - "8000:8000"
    volumes:
      - "./deployment:/deployment"
```
I can mount a local deployment directory with the `config.toml` in it.

the aformentioned error is originated by the[`_get_deployments` function](https://github.com/mamba-org/quetz/blob/master/quetz/cli.py#L264) that tries to create the directory in which to place the config file.

this file is only used to keep track of existing deployments, but I don't think it's useful in a docker context;
I suppose that the (non)existence check can be better performed by looking at the actual files and directories inside the deployment folder.

If it's ok with you I'm going to try to fix the issue by removing this file.

Maybe we should try to better define the cli commands, but I think it's already mentioned in #155

**Comment by sanzoghenzo** [comment]
Now that I better analyze the situation, the deployments.json is also used to allow and memorize different config file names.

Since the create command only allows a single config file in a deployment directory, and that the default is `config.toml`, isn't it better to just allow that filename?

To be able to have a different config name (and maybe store it in a central location), the --copy-conf could take any toml path and copy it to `config.toml` in the deployment folder.

**Comment by wolfv** [comment]
So far I have created the config.toml on the host machine, and loaded the config.toml in the docker / docker-compose file using `quetz start /path/to/mounted_folder/ ...`

That has worked reasonably well, but I can understand the idea of encapsulating quetz completely in a docker container.
We're very much open for improvements here :)

**Comment by sanzoghenzo** [comment]
Hi @wolfv,
this it exactly what I'm trying to do, have the deployment directory and config on the host and make the image as "frozen" as possible.

Do you mean that you use `docker run -v /host/folder:/path/to/config image_name quetz start path/to/config` or use the `command:` section in docker-compose, without having to use `quetz create` first?

If I do that, it complains about the directory not being a deployment.

BTW, the issue I was having was due to the user nobody that cannot create folders, but this highlighted the fact that the service is trying to write to another path (other than the expected one) just for a single file.

To be able to restart or upgrade the docker container and not lose the deployments.json file, one should bind mount also the home directory of the user running quetz;
in my case, the user didn't have any home (I was trying to avoid the groupadd and useradd commands in the Dockerfile), so the bind mount wasn't so easy 🤦 

Having said that, I still think that the file is not needed, even for the most complex scenario.
Maybe @adriendelsalle can shed some light on the reasoning behind this?

**Comment by adriendelsalle** [comment]
hi @sanzoghenzo and sorry for the delay.

Obviously I didn't tried to use Quetz that way (in a Docker container logged as a user without home directory).
Seems you fixed that with a PR, thank you !
